packages/compass-web/src/index.tsx

```diff
@@ -145,17 +108,17 @@ const CompassWeb = ({
   // testing purposes, should never be used otherwise
   __TEST_MONGODB_DATA_SERVICE_CONNECT_FN,
 }: CompassWebProps) => {
-  // TODO(COMPASS-7415): this should be supported by the preferences service /
-  // preferences provider
-  const preferences = useSimplePreferencesService({
-    maxTimeMS: 10_000,
-    enableExplainPlan: true,
-    enableAggregationBuilderRunPipeline: true,
-    enableAggregationBuilderExtraOptions: true,
-    enableImportExport: false,
-    enableSavedAggregationsQueries: false,
-    ...initialPreferences,
-  });
+  const preferencesAccess = useRef(
+    new ReadOnlyPreferenceAccess({
+      maxTimeMS: 10_000,
+      enableExplainPlan: true,
+      enableAggregationBuilderRunPipeline: true,
+      enableAggregationBuilderExtraOptions: true,
+      enableImportExport: false,
+      enableSavedAggregationsQueries: false,
+      ...initialPreferences,
+    })
+  );
```

Comment:
Here is where I'm still a bit unsure – what is the specific reason we're not creating an instance of `Preferences` with `preferencesStorage: new InMemoryStorage()`? I know that that would add support for some behaviors that are not strictly required, but it would still overall simplify the code a bit and make it more consistent, no?

Reply:
The ReadOnlyPreferenceAccess does exactly what you mentioned [here](https://github.com/mongodb-js/compass/pull/5356/files/9b5c4e2a37ec49a6bc138c5cb71a78b2371f3bbe#diff-e4235595e415da0655b882fa6d85530719e2601e39d15d6f6f03903df2077397R12-R15). This only acts as a service wrapper for actual Preferences. The reason I couldn't use Preferences directly as the value for provider is because there is an interface mismatch between PreferencesAccess and Preferences.